### DISCUSSION TITLE
Annotation(Dialog(enable = BooleanArray[index]))

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#11360] [Q&A]
Hi,

I wonder if the enable field annotation(dialog(enable = )) allows a boolean parameter array in Modelica standard. This would allow controlling the enable of several parameters as a function of the user selection without defining any other intermediate parameter, as shows in this code (paramSources is an enumeration type):

``` Modelica
block A
  parameter Interfaces.paramSources indSourceSelect = Interfaces.paramSources.exp
      annotation(choicesAllMatching = true);
  parameter SI.Time period=120 
      annotation(Dialog(enable = sourceSelect[Interfaces.paramSources.trapezoid]));
protected
    parameter Boolean sourceSelect[Interfaces.paramSources] = 
      {if i == indSourceSelect then true else false for i in Interfaces.paramSources} "Array of conditional flags";
end A;
```

Thanks in advance!

**Comment by mpradovelasco** [comment]
A more important question is related to the fact that enable does not react to changes in a parameter index until the dialog is saved and reopen again. That is, if I select the following value for indSourceSelect with the drop-down list:

![imagen](https://github.com/OpenModelica/OpenModelica/assets/114144098/a2e30dd0-ef31-4f2e-84b1-574b40465e07)

this value is not considered in the enable until the OK buttom is selected and the dialog box for that component is reopened, despite the enable is set now as follows:

```Modelica
parameter SI.Time period=120 
      annotation(Dialog(enable = flag));
parameter Boolean flag = if indSourceSelect == Interfaces.paramSources.trapezoid then true else false;

```
I wonder if there is a method to achieve that enable react without push OK buttom. 

System: latest OM 1.23-dev and Windows 11.

Thanks again.

**Reply by casella** [reply]
@adeas31 can you trigger a refresh of the dialog as soon as you change any parameter, so that parameter-depended dialogs are updated right away?

**Comment by mpradovelasco** [comment]
I have created an issue associated with another scenario where enable reaction is delayed (I think it is a bug) in #11519.

**Comment by mpradovelasco** [comment]
Hi @casella  and @adeas31 ,

In my opinion, this issue has similar relevance with respect to take advantage of the GUI feedback that #11519 . 

Thanks again!

**Comment by casella** [comment]
@mpradovelasco, in principle I don't see why annotation(Dialog(enable = BooleanArray[index])) should not be allowed, as long as the RHS of the enable binding is a boolean scalar. Maybe what is currently not working is updating this index correctly. Please open a ticket about that.

**Comment by mpradovelasco** [comment]
OK @casella , I will open a ticket with a minimal model of this issue. 

Thanks for your effort!
